lib/cib/cib_utils.c

```diff
@@ -822,15 +822,19 @@ cib_internal_op(cib_t * cib, const char *op, const char *host,
                 const char *section, xmlNode * data,
                 xmlNode ** output_data, int call_options, const char *user_name)
 {
-    int (*delegate) (cib_t * cib, const char *op, const char *host,
-                     const char *section, xmlNode * data,
-                     xmlNode ** output_data, int call_options, const char *user_name) =
-        cib->delegate_fn;
+    int (*delegate)(cib_t *cib, const char *op, const char *host,
+                    const char *section, xmlNode *data, xmlNode **output_data,
+                    int call_options, const char *user_name) = NULL;
 
-    if(user_name == NULL) {
+    CRM_CHECK(cib != NULL, return -EINVAL);
```

Comment:
A public `cib_t:cib_api_operations_t` API call could get us here, for example:
```
cib->cmds->delete(NULL, ...)
```

In case you want to avoid the `CRM_CHECK()`. I look forward to these going away as public API but that's a long way off.

Reply:
true, a regular if makes sense